--- Makefile.orig       2012-09-04 12:56:52.000000000 -0700
+++ Makefile            2012-10-15 13:27:26.741588132 -0700
@@ -6,7 +6,7 @@
 .PHONY: default all opt install doc test
 default: all opt test_biniou META
 all: biniou.cma
-opt: biniou.cmxa bdump
+opt: biniou.cmxa biniou.cmxs bdump
 
 test: test_biniou
        ./test_biniou
@@ -45,6 +45,9 @@
        ocamlfind ocamlopt -a $(FLAGS) \
                -o biniou.cmxa -package "$(PACKS)" $(SOURCES)
 
+biniou.cmxs: biniou.cmxa
+       ocamlopt -shared -linkall -I . -o biniou.cmxs biniou.cmxa
+
 bdump: $(SOURCES) bdump.ml
        ocamlfind ocamlopt -o bdump $(FLAGS) \
                -package $(PACKS) -linkpkg \